src/nbt/compound.rs

```diff
@@ -113,3 +108,40 @@ impl NbtCompound {
             .and_then(|tag| tag.extract_long_array())
     }
 }
+
+impl From<Nbt> for NbtCompound {
+    fn from(value: Nbt) -> Self {
+        value.root_tag
+    }
+}
+
+impl FromIterator<(String, NbtTag)> for NbtCompound {
+    fn from_iter<T: IntoIterator<Item = (String, NbtTag)>>(iter: T) -> Self {
+        Self {
+            child_tags: HashMap::from_iter(iter),
+        }
+    }
+}
+
+impl IntoIterator for NbtCompound {
+    type Item = (String, NbtTag);
+
+    type IntoIter = IntoIter<String, NbtTag>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.child_tags.into_iter()
+    }
+}
+
+impl Extend<(String, NbtTag)> for NbtCompound {
+    fn extend<T: IntoIterator<Item = (String, NbtTag)>>(&mut self, iter: T) {
```

Comment:
`    type Item = (String, NbtTag);`
Above you declare this but here dont use it and instead making it redeclare? like okaaaaaaaaaay

Weird things being here clean it up